### PR TITLE
Remove Windows arm32 support

### PR DESCRIPTION
**Pull request by BruceForstall**
Mostly, remove the CI runs. Windows arm32 product support was removed before .NET 6.

### CODE REVIEW

**Comment by BruceForstall** [comment]
cc @dotnet/jit-contrib

**Comment by BruceForstall** [comment]
Comments, anyone?

There are probably many more things that could be removed after this is merged. The primary goal here was to stop running win-arm builds and tests in the CI system.

File: eng/pipelines/coreclr/templates/build-job.yml
Comment (jkoritzinsky) [inline eng/pipelines/coreclr/templates/build-job.yml:208]: There's a condition in `eng/Subsets.props` that corresponds to this condition. Do we want to remove that as well?

**Review by jkoritzinsky — APPROVED** [review]
LGTM. Do we want to remove support for building the product build for Windows ARM32, or just the runtime test build/run?

**Comment by jkotas** [comment]
There is more that can be cleaned up in the sources (e.g. delete src\coreclr\vm\arm\*.asm). It can be done as a separate change.

**Review by jkotas — APPROVED** [review]

File: src/tests/run.cmd
Comment (trylek) [inline src/tests/run.cmd:202]: Hmm, I'm still seeing linux_arm / linux_musl_arm jobs in the above pipelines, are we removing arm32 support even on these?

File: src/tests/issues.targets
Comment (trylek) [inline src/tests/issues.targets:294]: Not as part of this PR but I guess at some point we should audit this issue list and close those that are no longer relevant i.e. that only occur on the no longer supported Windows arm platform.

**Review by trylek — APPROVED** [review]
LGTM overall modulo one nitpick. I guess we'll likely want to make an equivalent change in the .NET 6 / 7 servicing branches, correct?

File: src/tests/run.cmd
Comment (BruceForstall) [reply src/tests/run.cmd:202]: No, linux-arm is still a supported platform. (But run.cmd is only used on Windows)

**Comment by BruceForstall** [comment]
> I guess we'll likely want to make an equivalent change in the .NET 6 / 7 servicing branches, correct?

Since we don't support win-arm in .NET 6/7, that would make sense.

File: eng/pipelines/coreclr/templates/build-job.yml
Comment (BruceForstall) [reply eng/pipelines/coreclr/templates/build-job.yml:208]: Yes. But that will be a follow-up.

**Comment by BruceForstall** [comment]
I decided to simplify this to only remove win-arm CI runs. Removing the ability to build via build scripting can happen later.

**Pull request merged**
